pkg/resolver/resolver.go

```diff
@@ -0,0 +1,61 @@
+package resolver
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/sonr-io/sonr/pkg/client"
+	"github.com/sonr-io/sonr/pkg/crypto/mpc"
+	"github.com/sonr-io/sonr/pkg/did"
+)
+
+type ResolverImpl struct {
+	wallet *mpc.Wallet
+	client *client.Client
+	did.Resolver
+}
+
+func New() did.Resolver {
+	return &ResolverImpl{}
+}
+
+// this could be abstracted into a provider for connections.
+// right now binded to the client pkg
+func (res *ResolverImpl) WithClient(client *client.Client) {
```

Comment:
nit: `WithClient` makes me think this method will return a `ResolverImpl`. `SetClient` would be more clear